server/connector.go

```diff
@@ -625,6 +647,107 @@ func (c *Connector) pumpConnections(ctx context.Context, frontendConn, backendCo
 	}
 }
 
+func (c *Connector) retryBackendConnection(ctx context.Context, backendHostPort string, nextState mcproto.State) (net.Conn, error) {
+	// We want to try to connect to the backend every backendRetryInterval
+	var backendTry retry.Backoff
+
+	// Set the retry timeouts based on the next state and autoscaler
+	switch nextState {
+	case mcproto.StateStatus:
+		// Status request: try to connect once with backendStatusTimeout
+		backendTry = retry.NewConstant(backendStatusTimeout)
+		backendTry = retry.WithMaxRetries(0, backendTry)
+	case mcproto.StateLogin:
+		backendTry = retry.NewConstant(backendRetryInterval)
+		// Connect request: if autoscaler is enabled, try to connect until backendTimeout is reached
+		if c.config.AutoScaleUp {
+			// Autoscaler enabled: retry until backendTimeout is reached
+			backendTry = retry.WithMaxDuration(backendTimeout, backendTry)
+		} else {
+			// Autoscaler disabled: try to connect once with backendRetryInterval
+			backendTry = retry.WithMaxRetries(0, backendTry)
+		}
+	default:
+		// Unknown state, return error
+		logrus.
+			WithField("backend", backendHostPort).
+			WithField("nextState", nextState).
+			Error("Unknown state, unable to connect to backend")
+		return nil, fmt.Errorf("unknown state: %d", nextState)
+	}
+
+	var backendConn net.Conn
+	if err := retry.Do(ctx, backendTry, func(ctx context.Context) error {
+		logrus.
+			WithField("backend", backendHostPort).
+			WithField("nextState", nextState).
+			Debug("Attempting to connect to backend")
+
+		var err error
+		backendConn, err = net.Dial("tcp", backendHostPort)
+		if err != nil {
+			return retry.RetryableError(err)
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return backendConn, nil
+}
+
+func (c *Connector) getFakeOnlineStatus(serverAddress string) *mcproto.StatusResponse {
+	// Try to get the status from the cache
+	status, hit := c.StatusCache.Get(serverAddress)
+	if !hit {
+		logrus.
+			WithField("serverAddress", serverAddress).
+			Debug("Failed to get status from cache, sending default status")
+
+		// If we can't get the status from the cache, send a default status
+		return &mcproto.StatusResponse{
+			Version: mcproto.StatusVersion{
+				Name:     "UNKNOWN",
+				Protocol: 0,
```

Comment:
Have you been able to see what the Minecraft client thinks of these version and protocol values?